src/osd/OSD.cc

```diff
@@ -10075,6 +10094,60 @@ void OSD::handle_conf_change(const ConfigProxy& conf,
   }
 }
 
+void OSD::maybe_override_max_osd_capacity_for_qos()
+{
+  // If the scheduler enabled is mclock, override the default
+  // osd capacity with the value obtained from running the
+  // osd bench test. This is later used to setup mclock.
+  if (cct->_conf.get_val<std::string>("osd_op_queue") == "mclock_scheduler") {
+    // Write 200 4MiB objects with blocksize 4KiB
+    int64_t count = 12288000; // Count of bytes to write
+    int64_t bsize = 4096;     // Block size
+    int64_t osize = 4194304;  // Object size
+    int64_t onum = 200;       // Count of objects to write
+    double elapsed = 0.0;     // Time taken to complete the test
+    stringstream ss;
+    dout(1) << __func__
+            << " osd bench test -"
+            << " max_bytes_count: " << count
+            << " blocksize: " << bsize
+            << " max_object_count: " << onum
+            << dendl;
+    int ret = run_osd_bench_test(count, bsize, osize, onum, &elapsed, ss);
+    if (ret != 0) {
+      dout(1) << __func__
+              << " osd bench err: " << ret
+              << " osd bench errstr: " << ss.str()
+              << dendl;
+      dout(1) << " Using default osd_mclock_max_capacity_iops" << dendl;
+    } else {
+      double rate = count / elapsed;
+      double iops = rate / bsize;
+      dout(1) << __func__
```

Comment:
I am not sure we need to log all the above lines at level 1 in this function. Only adding the following summary line to the osd log at level 1 and maybe cluster log as well, when we have successfully overridden the defaults, should be enough.

Reply:
'ceph config show' displaying it is good enough - no need to add it to the cluster log. agree about the osd log levels

Reply:
second, i'd suggest not put this in clog.

Reply:
I have used derr for the error case. As suggested, I just added a log at level 1 when the test succeeds.